service/controller/clusterapi/v29/resource/asgstatus/create.go

```diff
@@ -25,12 +26,18 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	var asgName string
 	{
-		i := &cloudformation.DescribeStackResourceInput{
-			LogicalResourceId: aws.String(key.RefWorkerASG),
-			StackName:         aws.String(key.StackNameTCCP(&cr)),
+		i := &ec2.DescribeInstancesInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String(fmt.Sprintf("tag:%s", key.TagCluster)),
+					Values: []*string{
+						aws.String(key.ClusterID(&cr)),
+					},
+				},
+			},
 		}
 
-		o, err := cc.Client.TenantCluster.AWS.CloudFormation.DescribeStackResource(i)
+		o, err := cc.Client.TenantCluster.AWS.EC2.DescribeInstances(i)
```

Comment:
Why finding ASG via instances when the ASG itself also has the tag?

Reply:
That was what I wanted to do but you cannot describe ASGs using tag filters. You need the name to describe them, which is what we need to describe them. That was part of my venting during standup. Either we use the Cloud Formation way, which breaks apparently in rare circumstances, or do something else. Going with the instances was the best I could come up with.

Reply:
Oh man... I feel you... Been there as well... Then I feel this is fine.